typescript/18-wordy/wordy.ts

```diff
@@ -0,0 +1,146 @@
+type Token =
+  | { type: 'OPERAND'; value: number }
+  | { type: 'OPERATOR' | 'UNKNOWN'; value: string };
+
+class Node {
+  operator: string;
+  leftOperand: Node | number | null;
+  rightOperand: Node | number | null;
+
+  constructor(operator: string, leftOperand: Node | number | null, rightOperand: Node | number | null) {
+    this.operator = operator;
+    this.leftOperand = leftOperand;
+    this.rightOperand = rightOperand;
+  }
+}
+
+function buildAST(tokens: Token[], priority: Record<string, number>): Node | number {
+  const tree: (Node | number)[] = [];
+  const output: (Token | number)[] = [];
+  const operators: Token[] = [];
+
+  if (tokens[0].type === 'UNKNOWN') {
+    throw new Error('Unknown operation');
+  }
+
+  for (let i = 0; i < tokens.length; i++) {
+    const token: Token = tokens[i];
+    const nextToken: Token = tokens[i + 1];
+
+    if (token.type === 'OPERAND') {
+      if (nextToken?.type === 'UNKNOWN') {
+        throw new Error('Unknown operation');
+      }
+
+      if (nextToken?.type === 'OPERAND') {
+        throw new Error('Syntax error');
+      }
+
+      output.push(token.value);
+    } else if (token.type === 'OPERATOR') {
+      if (nextToken?.type !== 'OPERAND') {
+        throw new Error('Syntax error');
+      }
+
+      while (
+        operators.length > 0 &&
+        priority[operators[operators.length - 1].value] >= priority[token.value]
+        ) {
+        output.push(operators.pop()!);
+      }
+      operators.push(token);
+    }
+  }
+
+  while (operators.length > 0) {
+    output.push(operators.pop()!);
+  }
+
+  for (const item of output) {
+    if (typeof item === 'number') {
+      tree.push(item);
+    } else if (item.type === 'OPERATOR') {
+      const right: number | Node = tree.pop()!;
+      const left: number | Node = tree.pop()!;
+      tree.push(new Node(item.value, left, right));
+    }
+  }
+
+  return tree[0];
```

Comment:
How can you be sure that tree has just one element?
(Also, should be named plural)